main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AtkCocoaHelper/AtkCocoaHelperMac.cs

```diff
@@ -624,9 +625,31 @@ public static AccessibilityElementProxy ButtonElementProxy ()
 			return new AccessibilityElementProxy (new RealAccessibilityElementButtonProxy ());
 		}
 
-		public static AccessibilityElementProxy TextElementProxy ()
-		{
-			return new AccessibilityElementProxy (new RealAccessibilityElementNavigableStaticTextProxy ());
+		public static AccessibilityElementProxy TextElementProxy (Func<string> contents,
+																  Func<int> numberOfCharacters,
+																  Func<int> insertionPointLineNumber,
+																  Func<AtkCocoa.Range, Rectangle> frameForRange,
+																  Func<int, int> lineForIndex,
+																  Func<int, AtkCocoa.Range> rangeForLine,
+																  Func<AtkCocoa.Range, string> stringForRange,
+																  Func<int, AtkCocoa.Range> rangeForIndex,
+																  Func<int, AtkCocoa.Range> styleRangeForIndex,
+																  Func<Point, AtkCocoa.Range> rangeForPosition,
+																  Func<AtkCocoa.Range> visibleCharacterRange)
+		{
+			return new AccessibilityElementProxy (new RealAccessibilityElementNavigableStaticTextProxy () {
+				Contents = contents,
```

Comment:
on all the lines below:

```suggestion
				Contents = contents ?? throw new ArgumentNullException (nameof(contents)),
```